src/main/kotlin/lotto/domain/selling/Rank.kt

```diff
@@ -0,0 +1,21 @@
+package lotto.domain.selling
+
+enum class Rank(val matchCount: Int, val prizeMoney: Int) {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    MISS(0, 0);
+
+    companion object {
+        private val ranks: HashMap<Int, Rank> = HashMap(values().associateBy { it.matchCount })
```

Comment:
인터페이스를 타입으로 사용하는 습관을 길러 두면 프로그램이 훨씬 유연해질 거예요. 인터페이스와 다형성에 대해 알아보고 아래의 글도 참고하면 좋을 것 같아요.
https://stackoverflow.com/questions/11683044/return-arraylist-or-list/11683073

```suggestion
        private val ranks: Map<Int, Rank> = HashMap(values().associateBy { it.matchCount })
```